### ISSUE TITLE
delete dml get mysql-to-kafka job stop

### ISSUE THREAD

**Issue by ColdWaterLW** [actiontech/dtle#581]
## Steps to reproduce the issue
1. create a mysql-to-kafka job
2. execute a delete dml on mysql instance after full copy

## Describe the results you received
the job stopped

## Describe the results you expected
the job runs normally
## Output of `./dtle version`:**

```
9.9.9.9-master-99b303c
```

## Additional information

the bug was introduced from dtle-3.21.01.0
related issue #518 

## Additional details (log, config, job config etc):
nomad.log

```
2021-02-05T09:46:07.767Z [DEBUG] client.driver_mgr.dtle: publish.after: driver=dtle @module=dtle.extractor gno=362 job=test_21_3307 n=1 timestamp=2021-02-05
T09:46:07.767Z
2021-02-05T09:46:07.814Z [ERROR] client.driver_mgr.dtle: onError: driver=dtle @module=dtle.extractor err="applier error: kafkaTransformDMLEventQueries: send
 msgs failed: length of topics, keys and values must be equal" job=test_21_3307 timestamp=2021-02-05T09:46:07.814Z
2021-02-05T09:46:07.814Z [DEBUG] client.driver_mgr.dtle: extractor shutdown: driver=dtle @module=dtle.extractor job=test_21_3307 timestamp=2021-02-05T09:46:
07.814Z
2021-02-05T09:46:07.814Z [INFO]  client.driver_mgr.dtle: extractor shutdown: driver=dtle job=test_21_3307 @module=dtle.extractor timestamp=2021-02-05T09:46:
07.814Z
2021-02-05T09:46:07.814Z [DEBUG] client.driver_mgr.dtle: BinlogReader.Close: driver=dtle job=test_21_3307 @module=reader timestamp=2021-02-05T09:46:07.814Z
2021-02-05T09:46:07.816Z [ERROR] client.driver_mgr.dtle: error GetEvent.: driver=dtle err="Sync was closed" job=test_21_3307 @module=reader timestamp=2021-0
2-05T09:46:07.816Z
```

**Comment by asiroliu** [comment]
# version:
```
3.21.01.0-3.21.01.0-382c535
```
# reproduce:
1. prepare date
```sql
mysql> create database test_1;
Query OK, 1 row affected (0.01 sec)
```
```bash
shell> sysbench /usr/share/sysbench/oltp_delete.lua --range_size=100 --table_size=2100 --tables=1 --mysql-host=172.100.9.1 --mysql-port=3307 --mysql-user=test --mysql-password=test --mysql-db=test_1 prepare
```
2. create job
```hcl
job "test1" {
  datacenters = ["dc1"]
  type = "service"

  group "Src" {
    affinity {
      attribute = "${node.unique.name}"
      value = "dtle-1"
      }
    task "src" {
      driver = "dtle"
      config {
        ReplicateDoDb = [{
          TableSchemaRegex = "test(.*)"
          TableSchemaRename = "test$${1}"
          Tables = [{
            TableName = "sbtest1"
          }]
        }]
        ConnectionConfig = {
          Host = "172.100.9.1"
          Port = 3307
          User = "test_src"
          Password = "test_src"
        }
      }
    }
  }
  group "Dest" {
    affinity {
      attribute = "${node.unique.name}"
      value = "dtle-2"
      }
		task "dest" {
      driver = "dtle"
      config {
        KafkaConfig = {
          Topic = "kafka1"
          Brokers = ["172.100.9.21:9092"]
          Converter = "json"
        }
      }
    }
  }
}
```
3. delete data
```sql
sysbench /usr/share/sysbench/oltp_delete.lua --range_size=100 --table_size=2100 --tables=1 --mysql-host=172.100.9.1 --mysql-port=3307 --mysql-user=test --mysql-password=test --mysql-db=test_1 --events=100 run
```
5. check dtle-1 log
```
2021-02-06T22:35:32.928+0800 [ERROR] client.driver_mgr.dtle: onError: driver=dtle err="applier error: kafkaTransformDMLEventQueries: send msgs failed: length of topics, keys and values must be equal" job=test1 @module=dtle.extractor timestamp=2021-02-06T22:35:32.927+0800
2021-02-06T22:35:32.928+0800 [DEBUG] client.driver_mgr.dtle: extractor shutdown: driver=dtle @module=dtle.extractor job=test1 timestamp=2021-02-06T22:35:32.928+0800
2021-02-06T22:35:32.928+0800 [INFO]  client.driver_mgr.dtle: extractor shutdown: driver=dtle job=test1 @module=dtle.extractor timestamp=2021-02-06T22:35:32.928+0800
2021-02-06T22:35:32.928+0800 [DEBUG] client.driver_mgr.dtle: publish.before: driver=dtle @module=dtle.extractor gno=10 job=test1 n=1 timestamp=2021-02-06T22:35:32.927+0800
2021-02-06T22:35:32.928+0800 [DEBUG] client.driver_mgr.dtle: publish: driver=dtle job=test1 len=303 subject=test1_incr_hete @module=dtle.extractor timestamp=2021-02-06T22:35:32.928+0800
2021-02-06T22:35:32.928+0800 [ERROR] client.driver_mgr.dtle: unexpected error on publish: driver=dtle err="nats: connection closed" job=test1 @module=dtle.extractor timestamp=2021-02-06T22:35:32.928+0800
2021-02-06T22:35:32.928+0800 [ERROR] client.driver_mgr.dtle: onError: driver=dtle @module=dtle.extractor err="nats: connection closed" job=test1 timestamp=2021-02-06T22:35:32.928+0800
2021-02-06T22:35:32.928+0800 [DEBUG] client.driver_mgr.dtle: BinlogReader.Close: driver=dtle @module=reader job=test1 timestamp=2021-02-06T22:35:32.928+0800
```

**Comment by asiroliu** [comment]
# version:
```
9.9.9.9-master-a0a374d
```
# reproduce:
1. prepare date
```sql
mysql> create database test_1;
Query OK, 1 row affected (0.01 sec)
```
```bash
shell> sysbench /usr/share/sysbench/oltp_delete.lua --range_size=100 --table_size=2100 --tables=1 --mysql-host=172.100.9.1 --mysql-port=3307 --mysql-user=test --mysql-password=test --mysql-db=test_1 prepare
```
2. create job
```hcl
job "test1" {
  datacenters = ["dc1"]
  type = "service"

  group "Src" {
    affinity {
      attribute = "${node.unique.name}"
      value = "dtle-1"
      }
    task "src" {
      driver = "dtle"
      config {
        ReplicateDoDb = [{
          TableSchemaRegex = "test(.*)"
          TableSchemaRename = "test$${1}"
          Tables = [{
            TableName = "sbtest1"
          }]
        }]
        ConnectionConfig = {
          Host = "172.100.9.1"
          Port = 3307
          User = "test_src"
          Password = "test_src"
        }
      }
    }
  }
  group "Dest" {
    affinity {
      attribute = "${node.unique.name}"
      value = "dtle-2"
      }
		task "dest" {
      driver = "dtle"
      config {
        KafkaConfig = {
          Topic = "kafka1"
          Brokers = ["172.100.9.21:9092"]
          Converter = "json"
        }
      }
    }
  }
}
```
3. delete data
```sql
sysbench /usr/share/sysbench/oltp_delete.lua --range_size=100 --table_size=2100 --tables=1 --mysql-host=172.100.9.1 --mysql-port=3307 --mysql-user=test --mysql-password=test --mysql-db=test_1 --events=100 run
```
5. check dtle-1 log
```
no errors
```
